base/src/distributed/comms_mpi_hostbuffer_stream.cu

```diff
@@ -1560,6 +1560,57 @@ void CommsMPIHostBufferStream<T_Config>::all_gather_v_templated(T &my_data, int
 #endif
 }
 
+template <class T_Config>
+void CommsMPIHostBufferStream<T_Config>::all_gather_v(HDVector& data, int num_elems, HDVector& gathered_data, HIVector counts, HIVector displs)
+{
+#ifdef AMGX_WITH_MPI
+    MPI_Allgatherv(data.raw(), num_elems, MPI_DOUBLE, gathered_data.raw(), counts.raw(), displs.raw(), MPI_DOUBLE, mpi_comm);
```

Comment:
I might have misunderstood but this all_gather_v is implemented to work on an existing host array, rather than working on device memory. I could adapt this to work with CUDA-aware MPI in the future, but would you mind that being a future optimisation rather than part of this patch?

Reply:
I believe this happens implicitly - as the `CommsMPIDirect` inherits from `CommsMPIHostBufferStream`. It's a precedent set by a number of other similar routines.

Reply:
Ah, ok then, i'm starting to forget some internals. Thanks for confirming!